### PR TITLE
[merged] Hard panic everywhere; scheduler timeout for pipeline run

### PR DESCRIPTION
**Pull request by samsondav**
Recovering panics in the pipeline can lead to inconsistent state and
hang the pipeline indefinitely. It is better to hard panic immediately
and allow the node to be cleanly restarted.

In addition, the pipeline runner should have some kind of sanity
timeout. This commit re-adds a deadline for JobPipelineMaxRunDuration
which appears to have been removed by accident at some point.

---

The second commit completely removes WrapRecover and adds hard
panics everywhere.

Recover-ing gives illusion of safety but in reality it makes things more
dangerous.

In almost all cases its safer to hard panic the node than try to recover
and keep going since panicking can leave things in an unknown state.

Node operators typically run with auto-restart enabled, if the node
panics it will be booted again from a clean state, and we immediately
get a clear stacktrace of where the problem is.

We have seen bugs that would have been uncovered much sooner and caused
fewer problems if we had not suppressed the panic.

There is one and only one case I can think of where recovering panics is
acceptable, and that is in database transactions where it can be
intercepted in order to rollback the transaction, then re-panicked
again - and even that must be done very carefully with an additional
panic added with a timeout in case the rollback hangs.

### CODE REVIEW

**Review by pinebit — APPROVED** [review]

**Review by jmank88 — DISMISSED** [review]

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 355225a6607dba7db5b64456e7c3b1df2bd0d285) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Comment by samsondav** [comment]
@cl-homu retry

**Comment by samsondav** [comment]
@cl-homu +r

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably dacfecdacff012308d3872156248549902dd7903) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/shutdown/channel.go [hunk 1]
```diff
@@ -7,6 +7,14 @@ import (
...
 )
 
+var HardPanic bool
+
+func init() {
+	if os.Getenv("ENABLE_HARD_PANIC") == "true" {
+		HardPanic = true
+	}
+}
+
```
Comment (jmank88) [inline core/shutdown/channel.go:17]: Unused now?
```suggestion
```

Comment (samsondav) [reply core/shutdown/channel.go:17]: nope, its used in this file

Comment (samsondav) [reply core/shutdown/channel.go:17]: https://github.com/smartcontractkit/chainlink/pull/5384/files#diff-ad95a85a6c5931fa1b71b6341ccb7dd9659775e7f03ac9ebf17616ae2c27efcbR48

Comment (jmank88) [reply core/shutdown/channel.go:17]: Is that `Panic()` method unused?

**Comment by samsondav** [comment]
@cl-homu retry

**Review by jmank88 — APPROVED** [review]

**Comment by cl-homu** [comment]
:hourglass: Trying commit 20a7772394a06b7f5af9a242b9268ab613933d77 with merge f0ab5f21d510378c90e14d3187613bc1a6e0ee08...

**Comment by cl-homu** [comment]
:sunny: Try build successful - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4166665800?check_suite_focus=true), [checks-lint](https://github.com/smartcontractkit/chainlink/runs/4166665842?check_suite_focus=true), [checks-npm](https://github.com/smartcontractkit/chainlink/runs/4166665568?check_suite_focus=true), [checks-operator](https://github.com/smartcontractkit/chainlink/runs/4166665625?check_suite_focus=true), [checks-prettier](https://github.com/smartcontractkit/chainlink/runs/4166665936?check_suite_focus=true), [checks-solidity](https://github.com/smartcontractkit/chainlink/runs/4166665567?check_suite_focus=true)
Build commit: f0ab5f21d510378c90e14d3187613bc1a6e0ee08 (`f0ab5f21d510378c90e14d3187613bc1a6e0ee08`)

**Comment by jmank88** [comment]
@cl-homu +r

**Comment by samsondav** [comment]
@cl-homu r+

**Comment by cl-homu** [comment]
:broken_heart: Test failed - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4167084493?check_suite_focus=true)

**Comment by jmank88** [comment]
@cl-homu retry

**Comment by samsondav** [comment]
@cl-homu retry

**Comment by cl-homu** [comment]
:hourglass: Testing commit 20a7772394a06b7f5af9a242b9268ab613933d77 with merge 947cb03f3fcbf682f40541d26e379a1f71795233...

**Comment by cl-homu** [comment]
:broken_heart: Test failed - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4167430710?check_suite_focus=true)

**Comment by jmank88** [comment]
@cl-homu retry

**Comment by cl-homu** [comment]
:hourglass: Testing commit 20a7772394a06b7f5af9a242b9268ab613933d77 with merge fc94364fa46bf0d7f36ab4407a124a4530539fa2...

**Comment by cl-homu** [comment]
:sunny: Test successful - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4167909445?check_suite_focus=true), [checks-lint](https://github.com/smartcontractkit/chainlink/runs/4167909311?check_suite_focus=true), [checks-npm](https://github.com/smartcontractkit/chainlink/runs/4167909552?check_suite_focus=true), [checks-operator](https://github.com/smartcontractkit/chainlink/runs/4167909177?check_suite_focus=true), [checks-prettier](https://github.com/smartcontractkit/chainlink/runs/4167909537?check_suite_focus=true), [checks-solidity](https://github.com/smartcontractkit/chainlink/runs/4167909180?check_suite_focus=true)
Approved by: samsondav
Pushing fc94364fa46bf0d7f36ab4407a124a4530539fa2 to develop...

**Pull request closed without merging**